samples/TagHelperSample.Web/TagHelpers/ConditionalCommentTagHelper.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+
+namespace TagHelperSample.Web
+{
+    [TargetElement("iecondition")]
+    public class ConditionalCommentTagHelper : TagHelper
+    {
+        [HtmlAttributeName("mode")]
+        public string Mode { get; set; }
```

Comment:
Actually, do we even need the `Mode` Why not transform the tag itself into the condition? Do we need to support both types of conditions (I'm unaware if they do something different)? If we're going for functionality here I don't see the value in supporting both.